GridList/GridList.jsx

```diff
@@ -0,0 +1,178 @@
+ * @prop rows	{number} - Height of the tile in number of grid cells.
+ * @prop style {object} - Override the inline-styles of the root element.
+ * @prop subtitle	{Node} - String or element serving as subtitle (support text).
+ * @prop subtitleStyle {object} - Override the inline-styles of the subtitle element.
+ * @prop title {Node} - Title to be displayed on tile.
+ * @prop titleBackground {string} - 'rgba(0, 0, 0, 0.4)'	Style used for title bar background. Useful for setting custom gradients for example
+ * @prop titlePosition {'top'|'bottom'} - Position of the title bar (container of title, subtitle and action icon).
+ * @prop titleStyle {object} - Override the inline-styles of the title element.
+ */
+class GridListTile extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-list-tile";
+    this._propsDict = {
+      actionIcon: "actionIcon",
+      actionPosition: "actionPosition",
+      children: "children",
+      cols: "cols",
+      containerElement: "containerElement",
+      rows: "rows",
+      subtitle: "subtitle",
+      subtitleStyle: "subtitleStyle",
+      title: "title",
+      titleBackground: "titleBackground",
+      titlePosition: "titlePosition",
+      titleStyle: "titleStyle",
+      actionHandler: "actionHandler"
+    };
+  }
+  createClassName(props) {
+    const baseClass = "mdc-grid-list-tile--";
+    const classes = [];
+
+    if (props[this._propsDict.cols]) {
+      classes.push(baseClass + "span-" + props[this._propsDict.cols]);
+    }
+
+    return classes.join(" ");
+  }
+  materialDom(props) {
+    const actionHandler = props.actionHandler || function() {};
+    const action = props.actionIcon ? (
+      <i class="mdc-grid-tile__icon material-icons" onClick={actionHandler}>
+        {props.actionIcon}
+      </i>
+    ) : (
+      ""
+    );
+    return (
+      <li
+        className="mdc-grid-tile"
+        ref={control => {
+          this.control = control;
+        }}
+        {...props}
+      >
+        <div class="mdc-grid-tile__primary">
+          <div class="mdc-grid-tile__primary-content">{props.children}</div>
+        </div>
+        {props.title ? (
+          <span class="mdc-grid-tile__secondary">
+            {action}
+            <span class="mdc-grid-tile__title">{props.title}</span>
+            {props.subtitle ? (
+              <span class="mdc-grid-tile__support-text">{props.subtitle}</span>
+            ) : (
+              ""
+            )}
+          </span>
+        ) : (
+          ""
+        )}
+      </li>
+    );
+  }
+  render() {
+    const element = super.render();
+    // remove the extra attributes used for customising this element - keep the DOM clean
```

Comment:
Good initiative, we'll actually be working on an easier solution for this as requested in #240.

Reply:
That is great. I will keep an eye on that and if it gets done before this is ready to merge, I'll definitely incorporate that change.